File: src/test/java/com/polykhel/billies/service/dto/SubcategoryDTOTest.java
package com.polykhel.billies.service.dto;

import static org.assertj.core.api.Assertions.assertThat;

import com.polykhel.billies.web.rest.TestUtil;
import org.junit.jupiter.api.Test;

class SubcategoryDTOTest {

    @Test
    void dtoEqualsVerifier() throws Exception {
        TestUtil.equalsVerifier(SubcategoryDTO.class);
        SubcategoryDTO subcategoryDTO1 = new SubcategoryDTO();
        subcategoryDTO1.setId(1L);
        SubcategoryDTO subcategoryDTO2 = new SubcategoryDTO();
        assertThat(subcategoryDTO1).isNotEqualTo(subcategoryDTO2);
        subcategoryDTO2.setId(subcategoryDTO1.getId());
        assertThat(subcategoryDTO1).isEqualTo(subcategoryDTO2);
        subcategoryDTO2.setId(2L);
        assertThat(subcategoryDTO1).isNotEqualTo(subcategoryDTO2);
        subcategoryDTO1.setId(null);
        assertThat(subcategoryDTO1).isNotEqualTo(subcategoryDTO2);
    }
}

File: src/test/kotlin/com/polykhel/billnext/service/dto/SubcategoryDTOTest.kt
package com.polykhel.billnext.service.dto

import com.polykhel.billnext.web.rest.equalsVerifier
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class SubcategoryDTOTest {

    @Test
    fun dtoEqualsVerifier() {
        equalsVerifier(SubcategoryDTO::class)
        val subcategoryDTO1 = SubcategoryDTO()
        subcategoryDTO1.id = 1L
        val subcategoryDTO2 = SubcategoryDTO()
        assertThat(subcategoryDTO1).isNotEqualTo(subcategoryDTO2)
        subcategoryDTO2.id = subcategoryDTO1.id
        assertThat(subcategoryDTO1).isEqualTo(subcategoryDTO2)
        subcategoryDTO2.id = 2L
        assertThat(subcategoryDTO1).isNotEqualTo(subcategoryDTO2)
        subcategoryDTO1.id = null
        assertThat(subcategoryDTO1).isNotEqualTo(subcategoryDTO2)
    }
}
